pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -227,7 +227,7 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
 func (r *SystemReconciler) reconcileS3Storage() error {
 	// Nothing for reconcile.
 	// Check all required fields exist
-	awsCredentialsSecretName := r.apiManager.Spec.System.FileStorageSpec.S3.AWSCredentials.Name
+	awsCredentialsSecretName := r.apiManager.Spec.System.FileStorageSpec.S3.ConfigurationSecretRef.Name
```

Comment:
System reconciler now only handles the new S3 type struct fields (as the migration should take care to make sure that is used)